onnxscript/type_annotation.py

```diff
@@ -146,3 +164,102 @@ def get_return_types(typeinfo: type | typing.Sequence[type]) -> typing.Sequence[
+    """
+    if pytype is None:
+        return list(ALL_TYPE_STRINGS)
+    if pytype is type(None):
+        return list(ALL_TYPE_STRINGS)
+    if pytype is onnx_types.TensorType:
+        return list(ALL_TYPE_STRINGS)
+    if isinstance(pytype, type) and issubclass(pytype, onnx_types.TensorType):
+        return [pytype.to_string()]
+    if isinstance(pytype, onnx_types.TensorType):
+        return [pytype.to_string()]
+    if isinstance(pytype, TypeVar):
+        constraints = pytype.__constraints__
+        if constraints:
+            return pytype_to_input_strings(Union.__getitem__(constraints))
+        bound = pytype.__bound__
+        if bound is None:
+            return list(ALL_TYPE_STRINGS)
+        return pytype_to_input_strings(bound)
+    if typing.get_origin(pytype) is Union:
+        options = []
+        subtypes = typing.get_args(pytype)
+        # A None type in a Union is equivalent to an optional type
+        is_optional = any(subtype is type(None) for subtype in subtypes)
+        for subtype in subtypes:
+            if subtype is type(None):
+                # Skip None type because we are handling it with is_optional
+                continue
+            if is_optional:
+                options += [
+                    *pytype_to_input_strings(subtype),
+                    *[f"optional({s})" for s in pytype_to_input_strings(subtype)],
+                ]
+            else:
+                options += pytype_to_input_strings(subtype)
+        # Remove duplicates
+        return sorted(set(options))
+    if typing.get_origin(pytype) in _LIST_CONSTRUCTORS:
+        subtypes = typing.get_args(pytype)
+        return [f"seq({s})" for s in pytype_to_input_strings(subtypes[0])]
+
+    raise ValueError(f"Unsupported type: {pytype}")
+
+
+def get_type_constraint_name(pytype: TypeAnnotationValue) -> Optional[str]:
+    """Returns the name of the type constraint for a given type annotation.
+
+    Args:
+        pytype: A type annotation.
+
+    Returns:
+        The name of the type constraint if it is a TypeVar.
+        - Prefixes the name with "Optional_" if the type annotation is Optional[TypeVar].
+        - Prefixes the name with "Sequence_" if the type annotation is a Sequence[].
+        - Returns None if the type annotation does not have a type constraint.
+    """
+    if isinstance(pytype, TypeVar):
+        return pytype.__name__
+    if typing.get_origin(pytype) is Union:
+        subtypes = typing.get_args(pytype)
+        if len(subtypes) == 2 and type(None) in subtypes:
+            for subtype in subtypes:
+                if isinstance(subtype, TypeVar):
+                    return f"Optional_{subtype.__name__}"
+    if typing.get_origin(pytype) in _LIST_CONSTRUCTORS:
+        subtypes = typing.get_args(pytype)
+        if len(subtypes) == 1 and isinstance(subtypes[0], TypeVar):
+            return f"Sequence_{get_type_constraint_name(subtypes[0])}"
```

Comment:
Nit: make lines 260 and 264 consistent? Either use recursive call in both places or neither? Recursion may be better in case the function is further refined in the future. Perhaps
```py
type_param_name = get_type_constraint_name(subtypes[0])
return None if type_param_name is None else "Sequence_" + type_param_name
```

Reply:
Updated